app/User.php

```diff
@@ -218,4 +221,36 @@ public function getSoloAttribute() {
 
         return $date;
     }
+    
+    /**
+     * When doing $user->token, return a valid access token or null if none exists
+     * 
+     * @return \League\OAuth2\Client\Token\AccessToken 
+     * @return null
+     */
+    public function getTokenAttribute()
+    {
+        if ($this->access_token === null) return null;
+        else {
+            $token = new AccessToken([
+                'access_token' => $this->access_token,
+                'refresh_token' => $this->refresh_token,
+                'expires' => $this->token_expires,
+            ]);
+
+            if ($token->hasExpired()) {
+                $token = VatsimOAuthController::updateToken($token);
+            }
+
+            // Can't put it inside the "if token expired"; $this is null there
+            // but anyway Laravel will only update if any changes have been made.
+            $this->update([
+                'access_token' => ($token) ? $token->getToken() : null,
+                'refresh_token' => ($token) ? $token->getRefreshToken() : null,
+                'token_expires' => ($token) ? $token->getExpires() : null,
+            ]);
+
+            return $token;
+        }
+    }
```

Comment:
Again, I'm not sure if this is necessary for now. We can always add it in the future if we need it.

Reply:
Yes, we do still need that method - it is part of the new auth sequence.